--- a/src/FAudio_internal.c
+++ b/src/FAudio_internal.c
@@ -649,10 +649,11 @@
 	while (decoded < *toDecode && voice->src.queued_buffer_count)
 	{
 		float *dst = voice->audio->decoded_audio + (decoded * voice->src.format->nChannels);
-		struct queued_buffer *buffer = &voice->src.queued_buffers[0];
+		struct queued_buffer *buffer;
 		uint32_t decode_count;
 
 		try_collect_unaligned_data(voice);
+		buffer = &voice->src.queued_buffers[0];
 
 		/* Start-of-buffer behavior */
 		start_buffer(voice, buffer);
