--- a/src/FAudio_internal.c
+++ b/src/FAudio_internal.c
@@ -731,7 +731,7 @@
 #endif
 		{
 			uint32_t block_offset = voice->src.curBufferOffset % samples_per_block;
-			const uint8_t *src = buffer->buffer.pAudioData;
+			const uint8_t *src = buffer->buffer.pAudioData + buffer->first_block_offset;
 
 			src += (voice->src.curBufferOffset / samples_per_block) * block_size;
 
@@ -744,7 +744,7 @@
 		{
 			FAudio_zero(
 				voice->audio->decoded_audio + (
-					decoded * voice->src.format->nChannels
+					(decoded + decode_count) * voice->src.format->nChannels
 				),
 				sizeof(float) * (
 					(EXTRA_DECODE_PADDING - decode_count) *
