--- a/dlls/xaudio2_7/xapo.c
+++ b/dlls/xaudio2_7/xapo.c
@@ -32,6 +32,7 @@
 #include <FAudioFX.h>
 
 WINE_DEFAULT_DEBUG_CHANNEL(xaudio2);
+WINE_DECLARE_DEBUG_CHANNEL(faudio);
 
 static XA2XAPOFXImpl *impl_from_IXAPO(IXAPO *iface)
 {
@@ -234,6 +235,17 @@
 {
     XA2XAPOFXImpl *This = impl_from_IXAPOParameters(iface);
     TRACE("%p, %p, %u\n", This, params, params_len);
+    if (TRACE_ON(faudio) && params_len == sizeof(FAudioFXReverbParameters))
+    {
+        const FAudioFXReverbParameters *p = params;
+        TRACE_(faudio)("reverb %p wet=%g delays=%u,%u,%u position=%u,%u,%u,%u diffusion=%u,%u EQ=%u,%u,%u,%u room=%g,%g,%g reflection=%g reverb=%g decay=%g density=%g size=%g\n",
+            This, p->WetDryMix, p->ReflectionsDelay, p->ReverbDelay, p->RearDelay,
+            p->PositionLeft, p->PositionRight, p->PositionMatrixLeft, p->PositionMatrixRight,
+            p->EarlyDiffusion, p->LateDiffusion, p->LowEQGain, p->LowEQCutoff,
+            p->HighEQGain, p->HighEQCutoff, p->RoomFilterFreq, p->RoomFilterMain,
+            p->RoomFilterHF, p->ReflectionsGain, p->ReverbGain, p->DecayTime,
+            p->Density, p->RoomSize);
+    }
     This->fapo->SetParameters(This->fapo, params, params_len);
 }
 
